src/view/layer-shell/layer-shell.cpp

```diff
@@ -494,6 +496,7 @@ void wayfire_layer_shell_view::map()
     {
         this->app_id = nonull(lsurface->namespace_t);
         wf::view_implementation::emit_app_id_changed_signal(self());
+        wf_layer_shell_manager::get_instance().arrange_unmapped_view(this);
```

Comment:
Seems very weird to arrange_unmapped_view in the map handler, I think you can remove this call.